tiledb/sm/query/query.cc

```diff
@@ -2347,24 +2318,34 @@ Status Query::set_subarray_unsafe(const Subarray& subarray) {
   return Status::Ok();
 }
 
-Status Query::set_subarray(const tiledb::sm::Subarray& subarray) {
-  auto query_status = status();
-  if (query_status != tiledb::sm::QueryStatus::UNINITIALIZED &&
-      query_status != tiledb::sm::QueryStatus::COMPLETED) {
+void Query::set_subarray(const tiledb::sm::Subarray& subarray) {
+  // As implemented, this silently fails for queries that are initilized but not
+  // completed. It is unclear if the comment in the `if` statement below is
+  // correct or relevent anymore.
+  if (status_ != tiledb::sm::QueryStatus::UNINITIALIZED &&
+      status_ != tiledb::sm::QueryStatus::COMPLETED) {
     // Can be in this initialized state when query has been de-serialized
     // server-side and are trying to perform local submit.
     // Don't change anything and return indication of success.
-    return Status::Ok();
+    return;
   }
 
-  // Set subarray
+  // If the query is initialized, check that it doesn't use dimension labels and
+  // the subarray doesn't use dimension labels.
+  if (status_ != QueryStatus::UNINITIALIZED &&
+      (uses_dimension_labels() || subarray.has_label_ranges())) {
+    throw QueryStatusException(
+        "Cannot set subarray; Setting a subarray on a query that uses "
+        "dimension labels is not supported.");
+  }
```

Comment:
possibly confusing message content,
as elsewhere, appears if query is uninitialized it can be set...
see comment on other set_subarray()